cmd/analyzer/parse_args.go

```diff
@@ -60,26 +60,52 @@ const (
 	allSubnetsDiff   = "diff_all_subnets"   // semantic diff of allSubnets analysis between two configurations
 )
 
-var supportedOutputFormats = map[string]bool{
-	JSONFormat:       true,
-	TEXTFormat:       true,
-	MDFormat:         true,
-	DRAWIOFormat:     true,
-	ARCHDRAWIOFormat: true,
-	DEBUGFormat:      true,
+var supportedOutputFormatsMap = map[string][]string{
+	JSONFormat:       {allEndpoints, allSubnets},
+	TEXTFormat:       {allEndpoints, allSubnets, singleSubnet, allEndpointsDiff, allSubnetsDiff},
+	MDFormat:         {allEndpoints, allEndpointsDiff, allSubnetsDiff},
+	DRAWIOFormat:     {allEndpoints},
+	ARCHDRAWIOFormat: {allEndpoints},
+	DEBUGFormat:      {allEndpoints},
 }
-var supportedAnalysisTypes = map[string]bool{
-	allEndpoints:     true,
-	allSubnets:       true,
-	singleSubnet:     true,
-	allEndpointsDiff: true,
-	allSubnetsDiff:   true,
+var supportedAnalysisTypesMap = map[string][]string{
+	allEndpoints:     {TEXTFormat, JSONFormat, MDFormat, DRAWIOFormat, ARCHDRAWIOFormat, DEBUGFormat},
+	allSubnets:       {TEXTFormat, JSONFormat},
+	singleSubnet:     {TEXTFormat},
+	allEndpointsDiff: {TEXTFormat, MDFormat},
+	allSubnetsDiff:   {TEXTFormat, MDFormat},
 }
 
-func getSupportedValuesString(supportedValues map[string]bool) string {
+var supportedOutputFormatsList = []string{
+	JSONFormat,
+	TEXTFormat,
+	MDFormat,
+	DRAWIOFormat,
+	ARCHDRAWIOFormat,
+	DEBUGFormat,
+}
+var supportedAnalysisTypesList = []string{
+	allEndpoints,
+	allSubnets,
+	singleSubnet,
+	allEndpointsDiff,
+	allSubnetsDiff,
+}
+
+func getsupportedAnalysisTypesMapString() string {
```

Comment:
Rename func `getsupportedAnalysisTypesMapString` to `getSupportedAnalysisTypesMapString`

Install in VScode a spell checker extension (it shows this warning: `"getsupported": Unknown word.`)
Also, fix other places it shows typos on this file.